lib/execution_engine2/utils/application_info.py

```diff
@@ -21,57 +25,79 @@ class AppInfo:
     module - the app's module, e.g. kb_uploadmethods.
     method - the SDK method the app will run, e.g. import_reads_from_staging
     application_module - the module containing the application. Under normal conditions this
-        will be the same as 'module'.
+        will be the same as 'module', if not None. Always supplied if 'application' is not None.
     application - the id of the application, e.g. import_fastq_interleaved_as_reads_from_staging.
         This is the name of the folder in the 'ui/narrative/methods' folder in the app repo
-        contining the spec files for the app.
+        contining the spec files for the app. May be None.
     """
 
-    def __init__(self, method: str, app_id: str, strict=True):
+    def __init__(self, method: str, app_id: Union[str, None] = None, strict=True):
         """
         Create the application information.
 
         method - the method name, e.g. kb_uploadmethods.import_reads_from_staging
-        app_id - the app name, either fully qualified (e.g.
-            kb_uploadmethods/import_fastq_interleaved_as_reads_from_staging or unqualified (e.g.
-            import_fastq_interleaved_as_reads_from_staging). If fully qualified, the module name
-            of the app (kb_uploadmethds in this example) must match the module name for the method.
+        app_id - the app name in the module/app_name format (e.g.
+            kb_uploadmethods/import_fastq_interleaved_as_reads_from_staging). For historical
+            reasons, this class will also accept only the module name or the module.app_name
+            format. In both cases the module name must match that given for the method argument.
```

Comment:
I wonder if we should be storing "kbparallels", "concierge" or "cli ee2 call" or something in that field

Reply:
I don't think so - IMO each field should have a particular meaning if present. If I'm a user, I don't want a field to have more than one meaning that I have to figure out from the contents. If we want to have a field where we store the source of the job call, we should add another field, again IMO.